src/experience/service.py

```
@@ -132,18 +168,198 @@ def _generate_tags(experience: Experience) -> str:
+        # Validate that returned tags are from available tags
+        selected_tags = [tag.strip() for tag in tags.split(",")]
+        valid_tags = [tag for tag in selected_tags if tag in AVAILABLE_TAGS]
+
+        # Ensure 1-5 tags
+        if not valid_tags:
+            # Fallback: return empty list
+            return []
+        elif len(valid_tags) > 5:
+            valid_tags = valid_tags[:5]
+
+        return valid_tags
```

Ruff (0.14.14)

[warning] Consider moving this statement to an `else` block

(TRY300)